File: app/src/main/java/com/fly/cusheaderviewpager/MainActivity.java
package com.fly.cusheaderviewpager;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private TextView tvCustomLinearLayout;
    private Activity act;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        act = this;
        setContentView(R.layout.activity_main);

        tvCustomLinearLayout = findViewById(R.id.tv_custom_linearLayout);
        tvCustomLinearLayout.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        startActivity(new Intent(act, HeaderViewPagerActivity.class));
    }
}

File: app/src/main/java/com/fly/cusheaderviewpager/MainActivity.kt
package com.fly.cusheaderviewpager

import android.content.Intent
import android.view.View
import com.fly.basemodule.BaseActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity(), View.OnClickListener {
    override fun getLayoutId(): Int {
        return R.layout.activity_main
    }

    override fun initListeners() {
        super.initListeners()
        tv_custom_linearLayout.setOnClickListener(this)
    }


    override fun onClick(v: View) {
        startActivity(Intent(act, HeaderViewPagerActivity::class.java))
    }
}
